drivers/vmwarevsphere/cloudinit.go

```diff
@@ -0,0 +1,272 @@
+	}
+
+	return nil
+}
+
+func (d Driver) removeCloudInitIso(vm *object.VirtualMachine, dc *object.Datacenter, dss *object.Datastore) error {
+	c, err := d.getSoapClient()
+	if err != nil {
+		return err
+	}
+
+	path, err := d.getFolder(vm)
+	if err != nil {
+		return err
+	}
+
+	m := object.NewFileManager(c.Client)
+	task, err := m.DeleteDatastoreFile(d.getCtx(), dss.Path(filepath.Join(path, isoName)), dc)
+	if err != nil {
+		return err
+	}
+
+	if err = task.Wait(d.getCtx()); err != nil {
+		if types.IsFileNotFound(err) {
+			// already deleted, ignore error
+			return nil
+		}
+
+		return err
+	}
+
+	return nil
+}
+
+func (d Driver) createCloudInitIso() error {
+	//d.CloudConfig stat'ed and loaded in flag load.
+	sshkey, err := ioutil.ReadFile(d.publicSSHKeyPath())
+	if err != nil {
+		return err
+	}
+
+	userdatacontent, err := addSshUserToYaml(d.CloudConfig, d.SSHUser, string(sshkey))
+	if err != nil {
+		return err
+	}
+
+	perm := os.FileMode(0700)
+	isoDir := d.ResolveStorePath(isoDir)
+	dataDir := filepath.Join(isoDir, "data")
+	userdata := filepath.Join(dataDir, "user-data")
+	metadata := filepath.Join(dataDir, "meta-data")
+
+	err = os.MkdirAll(dataDir, perm)
+	if err != nil {
+		return err
+	}
+
+	writeYaml := fmt.Sprintf("#cloud-config\n%s", userdatacontent)
+	if err = ioutil.WriteFile(userdata, []byte(writeYaml), perm); err != nil {
+		return err
+	}
+
+	md := []byte(fmt.Sprintf("#local-hostname: %s\n", d.MachineName))
+	if err = ioutil.WriteFile(metadata, md, perm); err != nil {
+		return err
+	}
+
+	//making iso
+	blocksize := int64(2048)
+	diskImg := filepath.Join(isoDir, isoName)
+	iso, err := os.OpenFile(diskImg, os.O_CREATE|os.O_RDWR, perm)
+	if err != nil {
+		return err
+	}
+	defer iso.Close()
+
+	fs, err := iso9660.Create(iso, 0, 0, blocksize)
+	fs.Mkdir("/")
+	if err != nil {
+		return err
+	}
+
+	for filename, filepath := range map[string]string{"user-data": userdata, "meta-data": metadata} {
+		f, err := ioutil.ReadFile(filepath) // just pass the file name
+		if err != nil {
+			return err
+		}
+
+		rw, err := fs.OpenFile("/"+filename, os.O_CREATE|os.O_RDWR)
+		if err != nil {
+			return err
+		}
+
+		rw.Write(f)
+	}
+
+	if err = fs.Finalize(iso9660.FinalizeOptions{
+		RockRidge:        true,
+		VolumeIdentifier: "cidata",
+	}); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (d Driver) mountCloudInitIso(vm *object.VirtualMachine, dc *object.Datacenter, dss *object.Datastore) error {
+	log.Debugf("Mounting cloudinit %s", isoName)
+	devices, err := vm.Device(d.getCtx())
+	if err != nil {
+		return err
+	}
+
+	ide, err := devices.FindIDEController("")
+	if err != nil {
+		return err
+	}
+
+	var add []types.BaseVirtualDevice
+	cdrom, err := devices.CreateCdrom(ide)
+	if err != nil {
+		return err
+	}
+
+	path, err := d.getFolder(vm)
+	if err != nil {
+		return err
+	}
+
+	add = append(add, devices.InsertIso(cdrom, dss.Path(filepath.Join(path, isoName))))
+	if vm.AddDevice(d.getCtx(), add...); err != nil {
```

Comment:
You can just return the call instead of the if check

Reply:
done